src/seedu/addressbook/commands/Command.java

```diff
@@ -43,6 +43,14 @@ public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
 
+    /**
+     * Indicate if a command is mutable;
+     */
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implement in child classes");
```

Comment:
Good effort in trying to force the child classes to implement their own method definition.

Perhaps we can make this an abstract method? It will be a cleaner solution as the IDE will alert you instantly if the child classes do not implement `isMutating`.